File: 0_Temp/devide/d.cpp
#include <iostream>
#include <cmath>

double roundHalfToEven(double num) {
    double roundedDown = std::floor(num);
    double roundedUp = std::ceil(num);
    double rounded = std::round(num);

    if (std::abs(rounded - num) == 0.5 && std::fmod(roundedDown, 2.0) == 0.0) {
        return roundedDown;
    }

    return rounded;
}

int main() {
    double num1 = 2.5;
    double num2 = 3.5;

    double rounded1 = roundHalfToEven(num1);
    double rounded2 = roundHalfToEven(num2);

    std::cout << rounded1 << std::endl;  // Output: 2.0
    std::cout << rounded2 << std::endl;  // Output: 4.0

    return 0;
}

File: 0_Temp/devide/D.java
import java.math.*;

public class D {
    public static void main(String args[]) {
        double num1 = 2.5;
        double num2 = 3.5;

        long rounded1 = Math.round(num1);
        long rounded2 = Math.round(num2);

        System.out.println(rounded1);// Output: 3.0
        System.out.println(rounded2);// Output: 4.0
    }
}